src/main/java/io/cdap/plugin/gcp/dataplex/source/config/DataplexBatchSourceConfig.java

```diff
@@ -0,0 +1,412 @@
+      collector.addFailure("'Partition From Date' must be before or equal 'Partition To Date'.", null)
+        .withConfigProperty(BigQuerySourceConfig.NAME_PARTITION_FROM)
+        .withConfigProperty(BigQuerySourceConfig.NAME_PARTITION_TO);
+    }
+  }
+
+  /**
+   * Validate BigQuery Entity table validations
+   *
+   * @param collector
+   * @param project
+   * @param dataset
+   * @param tableId
+   */
+  public void validateTable(FailureCollector collector, String project, String dataset, String tableId) {
+    ConfigUtil.validateConnection(this, false, connection, collector);
+    BigQueryUtil.validateTable(tableId, NAME_ENTITY, collector);
+    if (containsMacro(NAME_LOCATION) || containsMacro(NAME_LAKE) || containsMacro(NAME_ZONE) ||
+      containsMacro(NAME_ENTITY)) {
+      return;
+    }
+    if (canConnect()) {
+      TableDefinition.Type definition = getSourceTableType(project, dataset, tableId);
+      if (definition != null && definition == TableDefinition.Type.VIEW) {
+        collector.addFailure(
+          String.format("'%s' is a 'View' :", tableId),
+          "In order to enable query views, please enable 'Enable Querying Views'");
+      }
+    }
+  }
+
+  /**
+   * Returns true if bigquery table can be connected and schema is not a macro.
+   */
+  private boolean canConnect() {
+    return !containsMacro(NAME_SCHEMA) &&
+      connection != null && connection.canConnect();
+  }
+
+  /**
+   * Reads table definition type from BigQuery
+   *
+   * @return {@link TableDefinition.Type}
+   */
+  public TableDefinition.Type getSourceTableType(String datasetProject, String dataSet,
+                                                 String tableId) {
+    Table sourceTable =
+      BigQueryUtil.getBigQueryTable(datasetProject, dataSet, tableId, getServiceAccount(), isServiceAccountFilePath());
+    return sourceTable != null ? sourceTable.getDefinition().getType() : null;
+  }
+
+
+  /**
+   * Setting validating input format as we are setting same format in task output format.
+   *
+   * @param pipelineConfigurer
+   * @param collector
+   * @param entity             Dataplex entity
+   */
+  public void setupValidatingInputFormat(PipelineConfigurer pipelineConfigurer, FailureCollector collector,
+                                         @Nullable Entity entity) {
+    ConfigUtil.validateConnection(this, false, connection, collector);
+    String fileFormat = INPUT_FORMAT;
+    collector.getOrThrowException();
+    if (entity != null) {
+      if (this.getSchema(collector) == null) {
+        Schema schema = DataplexUtil.getTableSchema(entity.getSchema(), collector);
+        pipelineConfigurer.getStageConfigurer().setOutputSchema(schema);
+      }
+    }
+    PluginProperties.Builder builder = PluginProperties.builder();
+    builder.addAll(this.getRawProperties().getProperties());
+    pipelineConfigurer.usePlugin("validatingInputFormat", fileFormat, fileFormat,
```

Comment:
So we will always set the avro input format no matter the entity is bigquery or GCS right?

Reply:
this method will always get called when it is a GCS entity.